source/plugin/advanced/debugging.rst

```diff
@@ -0,0 +1,119 @@
+================================
+Plugin Debugging and Hotswapping
+================================
+
+Every developer, at some point, has had the *pleasure* of debugging their plugin.
+Generally speaking, debugging is done by throwing a multitude of logs in specific areas to narrow down breakpoints,
+compiling the plugin, moving the result to the test folder, and running the server. Along that process, praying to
+whatever deity you have that everything doesn't immediately fail and make the past hour or so of programming in vain.
+
+However, what many newer programmers (and some seasoned!) don't know, is that there is an easier way. Why bother
+compiling and moving folders, when you can test your plugin from right within your IDE? Oh, of course you know how
+to run tests during compilation, but that's not quite what I'm talking about. We're talking about running the entire
+server implementation and plugin at once, from within the IDE.
+
+How is that possible, you may be asking? Quite simple. Plugin Debugging is fairly easy to setup, once you know how. And,
+with Plugin Debugging, comes Hotswapping (which we'll cover later).
+
+.. tip:: See the `Setting Up Your Workspace <../basics/workspace>`_ documentation, before attempting to follow this guide.
+
+Preparing your Plugin
+=====================
+
+IntelliJ
+~~~~~~~~
+
+* Open the **Sponge(Vanilla)** project.
+* Click ``File``, followed by ``New``, then ``Module...``.
+* Select ``Gradle`` in the popup, and click Next.
+* After naming your new project, click ``Finish``.
+
+Eclipse
+~~~~~~~
+
+.. tip:: You can use `Setting Up Eclipse <../basics/workspace/eclipse.rst>`_ to create a new Eclipse project. Eclipse, unlike IntelliJ, does not require any specific setup in regards to how you manage the plugin. Creating a new project is all you need to do.
+
+Adding Plugin to Sponge classpath
+=================================
+
+The idea behind this is that we'll launch Sponge from within your IDE, like normal. However, the difference is that
+we'll be adding your plugin to the classpath. To do this should only take a few steps, and will enhance your workflow,
+having cut down on all of the time that you previously would have spent transferring files and manually starting your
+server.
+
+.. tip:: From the Sponge `README.md <https://github.com/SpongePowered/Sponge/blob/master/README.md>`_, ensure that you have your Run/Debug Configuration set appropriately.
```

Comment:
Quibble. This would be clearer as "Examine the Sponge README.md <...>, and ensure"...  or similar. Or possibly reverse context, ie. "please ensure you have blahblah set appropriately, as shown in the Sponge README,md".